src/librustc/dep_graph/graph.rs

```diff
@@ -948,8 +940,8 @@ struct DepNodeData {
 }
 
 pub(super) struct CurrentDepGraph {
-    data: IndexVec<DepNodeIndex, DepNodeData>,
-    node_to_node_index: FxHashMap<DepNode, DepNodeIndex>,
+    data: Lock<IndexVec<DepNodeIndex, DepNodeData>>,
+    node_to_node_index: Lock<FxHashMap<DepNode, DepNodeIndex>>,
```

Comment:
Could you add a comment about locking order here? I.e. that `node_to_node_index` is always locked first (if both are locked) and that it is the source of truth for what nodes exist in the graph.